components/processor/gcomapiprocessor/internal/gcom/cache/auth_cache.go

```diff
@@ -0,0 +1,96 @@
+package cache
+
+import (
+	"context"
+	"errors"
+	"sync"
+	"time"
+
+	"github.com/go-kit/log"
+	"github.com/go-kit/log/level"
+	"github.com/grafana/opentelemetry-collector-components/processor/gcomapiprocessor/internal/gcom/client"
+	"github.com/opentracing/opentracing-go"
+)
+
+var errBadKey = errors.New("API Key Not Valid")
+
+type AuthCache struct {
+	client   client.Client
+	logger   log.Logger
```

Comment:
The collerctor uses zap instead -- does it make sense to switch to it at a follow-up PR?